### DISCUSSION TITLE
Need help understanding registry concept

### DISCUSSION THREAD

**Discussion by ituri** [netdata/netdata#16737] [Q&A]
Hi,

I'm trying to run my own registry for all of my netdata nodes. I followed to instructions at https://learn.netdata.cloud/docs/configuring/securing-netdata-agents/registry#run-your-own-registry and added these lines on the netdata node that I want to act as the registry:
`[registry]
        enabled = yes
        registry to announce = http://100.74.xxx.xxx:19999
` 
As you might see, the IP address used is from a (tailscale) tailnet that all my nodes have access to. On the client node I added these lines:
`
[registry]
        enabled = no
        registry to announce = http://100.74.xxxx.xxx:19999
`
When I access the registry node, I however can't create any spaces. Clicking save after entering a name doesn't do anything. Also I cannot add any nodes:
![CleanShot 2024-01-07 at 09 30 03](https://github.com/netdata/netdata/assets/871297/98b65b6a-a499-4742-981a-27439fc5f5e2)
Why is that? From my understanding I should be able to add nodes, create spaces etc. once a netdata node is configured to act as a registry. Or am I understanding the concept of a registry node wrong?

ituri

**Comment by akhil4674** [comment]
Try streaming from your nodes , by configuring the stream.conf
